examples/nlp/ner_transformers.py

```diff
@@ -0,0 +1,346 @@
+"""
+Title: Named Entity Recognition using Transformers
+Author: [Varun Singh](https://twitter.com/varun_singh3)
+Date created: Jun 23, 2021
+Last modified: Jun 23, 2021
+Description: NER using the Transformers and data from CoNLL 2003 shared task.
+"""
+"""
+# Named Entity Recognition (NER) using Transformers
+"""
+
+"""
+## Introduction
+"""
+"""
+Named Entity Recognition (NER) is the process of identifying named entities in text.
+Example of named entities are: "Person", "Location", "Organization", "Dates" etc. NER is
+essentially a token classification task where every token is classified into one or more
+predetermined categories.
+
+In this exercise, we will train a simple Transformer based model to perform NER. We will
+be using the data from CoNLL 2003 shared task. For more information about the dataset,
+please visi https://www.clips.uantwerpen.be/conll2003/ner/. However, since obtaining this
+data requires an additional step of getting a free license, we will be using
+HuggingFace's datasets library which contains a processed version of this dataset.
+"""
+
+"""
+## Install the open source datasets library from Huggingface
+"""
+
+"""shell
+!pip3 install datasets
+
+# The script used to evaluate NER models
+!wget https://raw.githubusercontent.com/sighsmile/conlleval/master/conlleval.py
+"""
+
+import os
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from datasets import load_dataset
+from collections import Counter
+from conlleval import evaluate
+
+"""
+We will be using the transformer implementation from this fantastic example:
+https://keras.io/examples/nlp/text_classification_with_transformer/
+"""
+
+# The TransformerBlock layer
+
+
+class TransformerBlock(layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super(TransformerBlock, self).__init__()
+        self.att = tf.keras.layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=embed_dim
+        )
+        self.ffn = tf.keras.Sequential(
+            [
+                tf.keras.layers.Dense(ff_dim, activation="relu"),
+                tf.keras.layers.Dense(embed_dim),
+            ]
+        )
+        self.layernorm1 = tf.keras.layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = tf.keras.layers.LayerNormalization(epsilon=1e-6)
+        self.dropout1 = tf.keras.layers.Dropout(rate)
+        self.dropout2 = tf.keras.layers.Dropout(rate)
+
+    def call(self, inputs, training=False):
+        attn_output = self.att(inputs, inputs)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(inputs + attn_output)
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        return self.layernorm2(out1 + ffn_output)
+
+
+# The embedding layer
```

Comment:
Move this to a text sentence

Reply:
I changed it to

"""
## The embedding layer
"""

Let me know if this is not what you meant.